src/doxygen.cpp

```diff
@@ -12462,6 +12463,12 @@ void generateOutput()
     g_outputList->add<RTFGenerator>();
     RTFGenerator::init();
   }
+  if (!(generateHtml && generateLatex && generateDocbook && generateMan && generateRtf))
+  {
+    g_outputList->add<ValidatingGenerator>();
+    ValidatingGenerator::init();
+  }
+
```

Comment:
Would it be easier to do this instead?

```c++
  if (g_outputList->size()==0)
  {
    Doxygen::suppressDocWarnings=TRUE;
  } 
```

Reply:
No I don't think so.
Doxygen consists, unfortunately, over 2 types of output generators (HTML / RTF / LaTeX / Docbook / Man / ... versus XML / SQLite / ...) and a number of tests / updates are only done in the first type of generators and hence when none of these generators we need a dummy generator that sees to it that the tests / updates are executed.

It might be that the test could be reduced to `g_outputList->size()==0` (or even `!g_outputList->size()`) though as this would be safer for the future.